test/boost/commitlog_test.cc

```diff
@@ -947,3 +947,73 @@ SEASTAR_TEST_CASE(test_commitlog_deadlock_with_flush_threshold) {
         co_await log.clear();
     }
 }
+
+
+// Unit test version of dtest to verify we enforce limit.
+SEASTAR_TEST_CASE(test_commitlog_disk_limit) {
+    commitlog::config cfg;
+
+    constexpr auto max_size_mb = 20;
+    cfg.commitlog_segment_size_in_mb = max_size_mb;
+    // ensure total size per shard is not multiple of segment size.
+    cfg.commitlog_total_space_in_mb = 5 * max_size_mb * smp::count;
+    cfg.commitlog_sync_period_in_ms = 10;
+    cfg.reuse_segments = true;
+    cfg.allow_going_over_size_limit = false;
+    cfg.use_o_dsync = true; // make sure we pre-allocate.
+
+    // not using cl_test, because we need to be able to abandon
+    // the log.
+
+    tmpdir tmp;
+    cfg.commit_log_location = tmp.path().string();
+    auto log = co_await commitlog::create_commitlog(cfg);
+
+    rp_set rps;
+    std::deque<rp_set> queue;
+    size_t n = 0;
+
+    // uncomment for verbosity
+    logging::logger_registry().set_logger_level("commitlog", logging::log_level::debug);
```

Comment:
This can be done externally when running the test.
Shouldn't be hard coded.

Reply:
removed